src/utilities/convex_hull.zig

```diff
@@ -721,19 +820,179 @@ test "Merge convex hull with single element's convex hull" {
     defer convex_hull_one.deinit();
 
     for (0..num_points) |i| {
-        try convex_hull_one.addPoint(.{ .time = i, .value = rnd.random().float(f64) });
+        try convex_hull_one.add(.{ .time = i, .value = rnd.random().float(f64) });
     }
 
     var convex_hull_two = try ConvexHull.init(allocator);
     defer convex_hull_two.deinit();
 
-    try convex_hull_two.addPoint(.{ .time = num_points, .value = rnd.random().float(f64) });
+    try convex_hull_two.add(.{ .time = num_points, .value = rnd.random().float(f64) });
 
-    try convex_hull_one.merge(&convex_hull_two);
+    try convex_hull_one.merge(&convex_hull_two, null);
 
     try testConvexHullProperty(&convex_hull_one);
 }
 
+test "Merge not-in-place convex hull with single element's convex hull" {
+    const num_points: usize = 100;
+    const allocator = testing.allocator;
+    var rnd = std.rand.DefaultPrng.init(0);
+
+    var convex_hull_one = try ConvexHull.init(allocator);
+    defer convex_hull_one.deinit();
+
+    for (0..num_points) |i| {
+        try convex_hull_one.add(.{ .time = i, .value = rnd.random().float(f64) });
+    }
+
+    var convex_hull_two = try ConvexHull.init(allocator);
+    defer convex_hull_two.deinit();
+
+    try convex_hull_two.add(.{ .time = num_points, .value = rnd.random().float(f64) });
+
+    var convex_hull_merged = try ConvexHull.init(allocator);
+    defer convex_hull_merged.deinit();
+
+    try convex_hull_one.merge(&convex_hull_two, &convex_hull_merged);
+
+    try testConvexHullProperty(&convex_hull_merged);
+}
+
+test "Merge not-in-place does not modify the convex hulls one and two" {
+    const num_points: usize = 100;
+    const allocator = testing.allocator;
+    var rnd = std.rand.DefaultPrng.init(0);
+
+    // Initialize convex_hull_one
+    var convex_hull_one = try ConvexHull.init(allocator);
+    defer convex_hull_one.deinit();
+    for (0..num_points) |i| {
+        try convex_hull_one.add(.{ .time = i, .value = rnd.random().float(f64) });
+    }
+
+    // Capture state of convex_hull_one before merge
+    const convex_hull_one_before = try convex_hull_one.getAllPointsSorted();
+    defer convex_hull_one_before.deinit();
+
+    // Initialize convex_hull_two
+    var convex_hull_two = try ConvexHull.init(allocator);
+    defer convex_hull_two.deinit();
+    for (num_points..2 * num_points) |i| {
+        try convex_hull_two.add(.{ .time = i, .value = rnd.random().float(f64) });
+    }
+
+    // Capture state of convex_hull_two before merge
```

Comment:
Before merging.

Reply:
fixed.

Reply:
Seems to still be wrong for `convex_hull_one` on Line 823.

Reply:
Got it. I think I got confused before.